profiling/src/internal/profile/fuzz_tests.rs

```diff
@@ -0,0 +1,631 @@
+    /// disk for the main binary and shared libraries, or virtual
+    /// abstractions like "[vdso]".
+    pub filename: Box<str>,
+
+    /// A string that uniquely identifies a particular program version
+    /// with high probability. E.g., for binaries generated by GNU tools,
+    /// it could be the contents of the .note.gnu.build-id field.
+    pub build_id: Box<str>,
+}
+
+impl Mapping {
+    pub fn new(
+        memory_start: u64,
+        memory_limit: u64,
+        file_offset: u64,
+        filename: Box<str>,
+        build_id: Box<str>,
+    ) -> Self {
+        Self {
+            memory_start,
+            memory_limit,
+            file_offset,
+            filename,
+            build_id,
+        }
+    }
+}
+
+impl<'a> From<&'a Mapping> for api::Mapping<'a> {
+    fn from(value: &'a Mapping) -> Self {
+        Self {
+            memory_start: value.memory_start,
+            memory_limit: value.memory_limit,
+            file_offset: value.file_offset,
+            filename: &value.filename,
+            build_id: &value.build_id,
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct Sample {
+    /// The leaf is at locations[0].
+    pub locations: Vec<Location>,
+
+    /// The type and unit of each value is defined by the corresponding
+    /// entry in Profile.sample_type. All samples must have the same
+    /// number of values, the same as the length of Profile.sample_type.
+    /// When aggregating multiple samples into a single sample, the
+    /// result has a list of values that is the element-wise sum of the
+    /// lists of the originals.
+    pub values: Vec<i64>,
+
+    /// label includes additional context for this sample. It can include
+    /// things like a thread id, allocation size, etc
+    pub labels: Vec<Label>,
+}
+
+impl<'a> From<&'a Sample> for api::Sample<'a> {
+    fn from(value: &'a Sample) -> Self {
+        Self {
+            locations: value.locations.iter().map(api::Location::from).collect(),
+            values: value.values.clone(),
+            labels: value.labels.iter().map(api::Label::from).collect(),
+        }
+    }
+}
+
+/// PartialEq and Hash work on just the key to avoid getting duplicate
+/// label keys.
+impl PartialEq for Label {
+    fn eq(&self, other: &Self) -> bool {
+        self.key == other.key
+    }
+}
+
+impl Eq for Label {}
+
+/// PartialEq and Hash work on just the key to avoid getting duplicate
+/// label keys.
+impl std::hash::Hash for Label {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        self.key.hash(state)
+    }
+}
+impl TypeGenerator for Sample {
+    fn generate<D: bolero_generator::Driver>(driver: &mut D) -> Option<Self> {
+        let locations = Vec::<Location>::gen_with().generate(driver)?;
+        let values = Vec::<i64>::gen_with().generate(driver)?;
+        let mut labels = std::collections::HashSet::<Label>::gen_with().generate(driver)?;
+
+        // Ensure that the label has "local root span id" key
```

Comment:
Why do we need to ensure this? Isn't it OK for a sample to not have a "local root span id"

Reply:
You're right, it's OK for a sample to not have it. However, I wasn't sure whether codes that deal with this logic would be covered?

Reply:
Take a boolean and add it iff the boolean is true?

Reply:
That's not possible as this is an implementation of a trait function.
https://docs.rs/bolero-generator/latest/bolero_generator/trait.TypeGenerator.html#tymethod.generate

I can try modifying what's generated by bolero using [mutate](https://docs.rs/bolero-generator/latest/bolero_generator/trait.TypeGenerator.html#method.mutate)?

Reply:
Also, `validate_sample_labels` doesn't allow us to add a label with `local root span id` key. Just deleting, this whole custom impl of `TypeGenerator`
 https://github.com/DataDog/libdatadog/blob/6f37a289241f85b7d379d5b8767925761dce09e4/profiling/src/internal/profile/mod.rs#L88